bm-persona/Data/Occupancy/Occupancy.swift

```diff
@@ -34,19 +34,23 @@ enum OccupancyStatus {
 
 // Object containing occupancy data for a location based on day of week and hour of day.
 class Occupancy {
-    
     // dictionary mapping day of week and hour of day to a percent occupancy
     var dailyOccupancy: [DayOfWeek: [Int: Int]]
+    // percent occupancy for the current time
+    var liveOccupancy: Int?
     // ranges for what percentage constitutes each status (high, medium, low)
     static let statusBounds: [OccupancyStatus: ClosedRange<Int>] = [OccupancyStatus.high: 70...100, OccupancyStatus.medium: 30...69, OccupancyStatus.low: 0...29]
-    //nice
     
-    init(dailyOccupancy: [DayOfWeek: [Int: Int]]) {
+    init(dailyOccupancy: [DayOfWeek: [Int: Int]], live: Int?) {
         self.dailyOccupancy = dailyOccupancy
+        self.liveOccupancy = live
     }
     
     // get a percent occupancy based on a date object (current day and time)
     func getOccupancyPercent(date: Date) -> Int? {
+        if let live = liveOccupancy {
+            return live
+        }
```

Comment:
This shouldn't return the live data if `date` is not near the current time, right? It looks like this function is only used to determine the status for the current time, but that might be unclear since it takes in a date parameter.

Maybe you could remove the date parameter to make it clear that this function should not be used to get occupancy data for other dates. If you want to add that capability maybe you could move the rest of the function body to `getHistoricalOccupancyPercent` or something along those lines.

Reply:
split into getCurrent and getHistoric (with current always using Date() and live data). also changed OccupancyGraphCardView to always use the current date because there were some inconsistencies/problems there.